### PR TITLE
[Feature Fix] Comparing email addresses during registration is now case insensitive

### PR DESCRIPTION
**Pull request by revanthkolli**
# Purpose
Regarding Issue #2902, changed the code so that the email comparison during registration is case insensitive. 

# Changes
In SignUp and views where the two email addresses inputed by the user were being compared, converted both to lowercase during the conversion so that case should not matter. 

# Side Effects
Doesn't affect the actual values for the emails, just uses the lowercase version of what the user input for comparison.

### CODE REVIEW

**Comment by samanehsan** [comment]
This looks great! Can you add a test case for this to `tests.test_views.TestAuthViews`? :umbrella:

**Comment by revanthkolli** [comment]
@samanehsan Sure!

**Comment by revanthkolli** [comment]
@samanehsan Added the test case.

**Comment by samanehsan** [comment]
Great! Can you bring your branch up to date with develop as well? Thanks!

**Comment by revanthkolli** [comment]
@samanehsan Yeah, forgot about that. :+1:

**Comment by samanehsan** [comment]
Thanks! @sloria, I think this is good to go! :confetti_ball:

File: framework/auth/views.py [hunk 1]
```diff
@@ -242,7 +242,7 @@ def register_user(**kwargs):
 
     """
     # Verify email address match
-    if request.json['email1'] != request.json['email2']:
+    if str(request.json['email1']).lower() != str(request.json['email2']).lower():
```
Comment (sloria) [inline framework/auth/views.py]: `request.get_json()` should be used here; `request.json` is deprecated: http://flask.pocoo.org/docs/0.10/api/#flask.Request.json.

I will make this change when I merge.

File: tests/test_views.py [hunk 2]
```diff
@@ -2520,6 +2520,21 @@ def test_register_ok(self):
         user = User.find_one(Q('username', 'eq', email))
         assert_equal(user.fullname, name)
 
+    def test_register_email_case_insensitive(self):
```
Comment (sloria) [inline tests/test_views.py]: When writing a regression test for code that addresses a GH issue, it is useful to include the URL for the issue in a code comment, e.g.

```python
    # Regression test for https://github.com/CenterForOpenScience/osf.io/issues/2902
    def test_register_email_case_insensitive(self):
```

Will add this when I merge.

**Pull request merged**